draft-ietf-cose-bls-key-representations.md

```diff
@@ -71,21 +71,19 @@ The following definitions apply to the pairing-friendly elliptic curves known as
 
 When expressing a cryptographic key for these curves in JSON Web Key (JWK) form, the following rules apply:
 
-- The parameter "kty" MUST be present and set to "EC".
+- The parameter "kty" MUST be present and set to "OKP".
 - The parameter "crv" MUST be present and value MUST be one defined in (#curve-parameter-registration).
-- The parameter "x" MUST be present whose value represents the x coordinate of the curve point for the public key. This value MUST be encoded using the procedure defined in [Section 2.1](#point-coordinates-encoding) and MUST be base64url encoded without padding as defined in [@!RFC7515] Appendix C.
-- The parameter "y" MUST be present whose value represents the y coordinate of the curve point for the public key. This value MUST be encoded using the procedure defined in [Section 2.1](#point-coordinates-encoding) and MUST be base64url encoded without padding as defined in [@!RFC7515] Appendix C.
-- The parameter "d" MUST be present for private key representations whose value MUST contain the little-endian representation of the private key base64url encoded without padding as defined in [@!RFC7515] Appendix C. This parameter MUST NOT be present for public keys.
+- The parameter "x" MUST be present with its value being the base64url encoding of the compressed serialized point format defined normatively in Appendix B of <xref target="BBS"/>.
+- The parameter "d" MUST be present for private key representations with its value being the base64url encoding of the compressed serialized point format defined normatively in Appendix B of <xref target="BBS"/>. This parameter MUST NOT be present for public keys.
```

Comment:
Neither BBS or BLS currently define encoding for the secret key. BBS also exclusively uses big-endian representations otherwise, while today `d` is little-endian (to match the Ed25519/Ed448 native encoding). BBS needs to represent points in its protocol, but not secrets.

The updated text also misrepresents the value of `d` to be a compressed point rather than a scalar.

I would suggest preserving the existing `d` parameter text here and in COSE

Reply:
OK - I'll apply your suggestion later this morning